### PR TITLE
Group 2

### PR DESCRIPTION
**Pull request by johanneskross**

### CODE REVIEW

File: src/main/java/Airport.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import models.MilitaryType;
+import Planes.MilitaryPlane;
+import Planes.Plane;
+
+import java.util.*;
+
+public class Airport {
```
Comment (heidekrueger) [inline src/main/java/Airport.java:7]: Is this the most appropriate name for the class?

File: src/main/java/Airport.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * Prints transport military planes
```
Comment (delsner) [inline src/main/java/Airport.java:10]: Duplication of information in comment and function name.
Incomplete description of provided functionality (prints the maximum capacity of all military planes)

File: src/main/java/Airport.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+     */
+    private void printMaxTransportMilitaryPlane() {
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
```
Comment (heidekrueger) [inline src/main/java/Airport.java:13]: this Array is never used.

File: src/main/java/Airport.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        Plane maxPlane = null;
```
Comment (delsner) [inline src/main/java/Airport.java:14]: In modern Java, this could be `Optional<Plane>`

File: src/main/java/Airport.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        // print banner
```
Comment (delsner) [inline src/main/java/Airport.java:16]: this seems obvious

File: src/main/java/Airport.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println ("*********************************************");
```
Comment (delsner) [inline src/main/java/Airport.java:17]: whitespace between `println (`

File: src/main/java/Airport.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println ("***** Largest Transport Military Plane ******");
+        System.out.println ("*********************************************");
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
```
Comment (delsner) [inline src/main/java/Airport.java:24]: `maxPlane == null || maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()`

File: src/main/java/Airport.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+                        maxPlane = plane;
+                    }
+                }
+            } //if
```
Comment (delsner) [inline src/main/java/Airport.java:28]: unnecessary comments here (if, else, for)

File: src/main/java/Airport.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            else {
+
+            } // else
+        } //for
+        System.out.println("model:" + maxPlane.getModel());
```
Comment (delsner) [inline src/main/java/Airport.java:33]: check for null pointer before accessing instance methods

File: src/main/java/Airport.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println("capacity:" + maxPlane.getMaxLoadCapacity());
+    }
+
+    //Constructor
```
Comment (delsner) [inline src/main/java/Airport.java:37]: unnecessary

File: src/main/java/Airport.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    public Airport(List<? extends Plane> planes) {
```
Comment (delsner) [inline src/main/java/Airport.java:38]: generic wildcard seems unnecessary, why not go with `List<Plane> planes`?

**Pull request closed without merging**